aws-resources/src/main/java/io/opentelemetry/contrib/aws/resource/EcsResource.java

```diff
@@ -99,6 +99,31 @@ static void fetchMetadata(
     }
   }
 
+  @Nullable
+  private static String getAccountId(@Nullable String arn) {
+    return getArnPart(arn, 4);
+  }
+
+  @Nullable
+  private static String getRegion(@Nullable String arn) {
+    return getArnPart(arn, 3);
+  }
+
+  @Nullable
+  private static String getArnPart(@Nullable String arn, int partIndex) {
+    if (arn == null) {
```

Comment:
```suggestion
    if (arn == null || partIndex < 0) {
```

Reply:
Pretty overkill if you ask me, but I can make a private enum out of it

Reply:
Introduced enum for ArnParts in 367c715

Reply:
Thanks, and I did agree that it's a little silly. :) Appreciate the extra mile tho.